### PR TITLE
[feat-4990]: Add accessibility to dashboard filter

### PR DESCRIPTION
**Pull request by OscarBakker**
**context**
Add custom accessibility to the dashboard filter.

**testers**
Should add on first render a tabindex on the logo in the header, cause the tabindex gets priority other the normal focusable items like links.

Space and arrow up /down should open a combobox(filter option) and bring you to the first or selected filter option. then navigate with arrows up and down, select option with enter / space, close dropdown with Escape.

Ticket: [SIG-4990](https://gemeente-amsterdam.atlassian.net/browse/SIG-4990)

## Signalen

Before opening a pull request, please ensure:

- Make sure your PR title follows naming conventions: [feat-1234]: name feature
- Double-check your branch is based on `main` and targets `main`
- Pull request has tests (we are going for 100% coverage!)
- Code is well-commented, linted and follows project conventions
- Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: src/signals/incident-management/containers/Dashboard/components/Filter/Option.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          }}
+          onClick={() => {
+            onChange(option)
+            setFilterNameActive('')
+          }}
+          onKeyDown={(e) => {
+            e.preventDefault()
+            setFocus(index)
+
+            if (['Enter', 'Space'].includes(e.code)) {
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/Dashboard/components/Filter/Option.tsx:61]: Looks like "enter" should open the dropdown as well right? Doesn't do so currently.

File: src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx [hunk 2]
```diff
@@ -258,17 +258,17 @@ describe('FilterComponent', () => {
...
     ).toHaveFocus()
 
     act(() => {
       fireEvent.keyDown(
-        screen.getByRole('listbox', {
+        screen.getByRole('combobox', {
           name: 'Categorie',
         }),
-        { code: 'Enter' }
+        { code: 'Space' }
```
Comment (vdegraaf) [inline src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx:271]: Why does "enter" not work anymore?

**Review by vdegraaf — COMMENTED** [review]
Nice!
Small comment:

- Test coverage of the new hook does not pass the branch threshold. 
Screenshot 2023-02-01 at 15 15 28

File: src/signals/incident-management/containers/Dashboard/components/Filter/Option.tsx (hunk 1 above)
Comment (OscarBakker) [reply src/signals/incident-management/containers/Dashboard/components/Filter/Option.tsx:61]: hmm no its should not, selecting is possible but opening not

File: src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx (hunk 2 above)
Comment (OscarBakker) [reply src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx:271]: its shouldn't check maar https://codepen.io/piyush-tapaniya/pen/MdNJNy

Comment (vdegraaf) [reply src/signals/incident-management/containers/Dashboard/components/Filter/Filter.test.tsx:271]: It doesn't either in the sandbox

**Pull request merged**